pandas/core/arrays/datetimelike.py

```diff
@@ -347,21 +326,59 @@ def _validate_frequency(cls, index, freq, **kwargs):
     # Arithmetic Methods
 
     def _add_datelike(self, other):
+        # Overriden by TimedeltaArray
         raise TypeError("cannot add {cls} and {typ}"
                         .format(cls=type(self).__name__,
                                 typ=type(other).__name__))
 
+    _add_datelike_dti = _add_datelike
```

Comment:
can you add either doc-strings or some expl here of what is going on here. Are you overridng _dti in the sub-classes? this seems like adding a lot of verbiage here.

Reply:
> Are you overridng _dti in the sub-classes?

Yes